vitrina/structure/views.py

```diff
@@ -3424,34 +3435,212 @@ def form_valid(self, form):
+                                old_base_metadata_instance
+                            )
+                            new_base_metadata_instance.object = new_base_related_instance
+                            new_base_metadata_instance.save()
+                            old_to_new_metadata_object_map[old_base_related_instance] = new_base_related_instance
+
+                    # Assign a duplicated related object to metadata
+                    new_metadata_instance.object = new_related_instance
+                    new_metadata_instance.save()
+
+                    old_to_new_metadata_object_map[old_related_instance] = new_related_instance
+
+                    # TODO: remove and calculate diff by using metadata table
+                    MetadataVersion.objects.create(
+                        metadata=new_metadata_instance,
+                        version=self.new_version,
+                        name=new_metadata_instance.name if new_metadata_instance.name else None,
+                        type=new_metadata_instance.type if new_metadata_instance.type else None,
+                        required=new_metadata_instance.required,
+                        unique=new_metadata_instance.unique,
+                        type_args=new_metadata_instance.type_args if new_metadata_instance.type_args else None,
+                        ref=new_metadata_instance.ref if new_metadata_instance.ref else None,
+                        source=new_metadata_instance.source if new_metadata_instance.source else None,
+                        prepare=new_metadata_instance.prepare if new_metadata_instance.prepare else None,
+                        level_given=new_metadata_instance.level_given,
+                        access=new_metadata_instance.access,
+                        base=new_metadata_instance.object.base
+                        if isinstance(new_metadata_instance.object, Model)
+                        else None,
+                        status=new_metadata_instance.status if new_metadata_instance.status else None,
+                    )
+
+            already_created_fields = old_to_new_metadata_object_map
+            for old_related_instance, new_related_instance in list(old_to_new_metadata_object_map.items()):
+                try:
+                    already_created_fields = self.duplicate_foreign_key_relationships(
+                        new_related_instance, already_created_fields
+                    )
+                except ValidationError as e:
+                    transaction.set_rollback(True)
+                    form.add_error(None, e.message)
+                    return self.form_invalid(form)
+
+        if self.new_version:
+            return redirect(reverse("dataset-structure", args=[self.dataset.pk, self.new_version.pk]))
+        else:
+            return redirect(reverse("dataset-structure", args=[self.dataset.pk, self.metadata_version.pk]))
```

Comment:
I'd set the second argument in a variable like:

`version_pk = self.new_version.pk if self.new_version else self.metadata_version.pk`

And use that as an argument:


```suggestion
        version_pk = self.new_version.pk if self.new_version else self.metadata_version.pk
        return redirect(reverse("dataset-structure", args=[self.dataset.pk, version_pk]))
```

Reply:
Nice, changed